### ISSUE TITLE
Failed test 'Dollar quotes with valid characters are parsed as identifiers'

### ISSUE THREAD

**Issue by akirpa** [bucardo/dbdpg#26]
Version 3.5.3 work w/o problem, but in the same environment
versions 3.5.9_1, 3.6.0 etc. fail during test with next messages:

```
t/12placeholders.t .. 1/257
#   Failed test 'Dollar quotes with valid characters are parsed as identifiers'
#   at t/12placeholders.t line 673.
#          got: 'Wide character in subroutine entry at /usr/i.kit/DBD-Pg-3.5.9_1/blib/lib/DBD/Pg.pm line 280.
# '
#     expected: ''

#   Failed test 'Dollar quotes with valid characters are parsed as identifiers'
#   at t/12placeholders.t line 673.
#          got: 'Wide character in subroutine entry at /usr/i.kit/DBD-Pg-3.5.9_1/blib/lib/DBD/Pg.pm line 280.
# '
#     expected: ''

#   Failed test 'Dollar quotes with valid characters are parsed as identifiers'
#   at t/12placeholders.t line 673.
#          got: 'Wide character in subroutine entry at /usr/i.kit/DBD-Pg-3.5.9_1/blib/lib/DBD/Pg.pm line 280.
# '
#     expected: ''
# Looks like you failed 3 tests of 257.
t/12placeholders.t .. Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/257 subtests
...
Test Summary Report
-------------------
t/12placeholders.t (Wstat: 768 Tests: 257 Failed: 3)
  Failed tests:  140-142
  Non-zero exit status: 3
Files=16, Tests=1918, 23 wallclock secs ( 0.27 usr  0.04 sys +  4.44 cusr  0.36 csys =  5.11 CPU)
Result: FAIL
Failed 1/16 test programs. 3/1918 subtests failed.
```

**Comment by ilmari** [comment]
@akirpa I cannot reproduce this problem. Could you please provide the output from `t/01connect.t`?

**Comment by akirpa** [comment]
@ilmari 

'Sorry for huge delay.
Problem exists in case SQL_ASCII encoding (in case UTF8 all work w/o problem). Please see below result of t/01connect.t in case problem:
---
# DBI                         Version 1.636
# DBD::Pg                     Version 3.6.2
# Perl                        Version 5.26.0
# OS                          freebsd
# PostgreSQL (compiled)       90507
# PostgreSQL (target)         90601
# PostgreSQL (reported)       PostgreSQL 9.6.1 on x86_64-unknown-freebsd10.2, compiled by clang version 3.8.1 (tags/RELEASE_381/final), 64-bit
# Default port                5432
# DBI_DSN                     dbi:Pg:
# DBI_USER                    postgres
# Test schema                 dbd_pg_testschema
# array_nulls                 on
# backslash_quote             safe_encoding
# client_encoding             UTF8
# server_encoding             SQL_ASCII
# standard_conforming_strings on
# Adjusted:                   DBI_DSN
---
Thanks in advance

**Comment by ilmari** [comment]
Thanks for the detail, that allowed me to hunt it down and fix it in commit 45e764b2d142fe151d51f00f1bd6e0278202ce45, which will be in the next release.
